.github/workflows/release.yml

```diff
@@ -21,33 +21,46 @@ jobs:
 
     steps:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           repository: wingtk/gvsbuild
           fetch-depth: 0
 
       - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v3
         with:
           python-version: ${{ matrix.python }}
           architecture: ${{ matrix.arch }}
 
       - name: Install Build Deps
-        run: python -m pip install wheel
+        run: python -m pip install wheel build
+
+      - name: Install Poetry
+        run: python -m pip install poetry==1.1.13
+      - name: Configure Poetry
+        run: poetry config virtualenvs.in-project true
+      - name: Install Python Dependencies
+        run: poetry install
 
       - name: Move git binary
         # Temporarily move preinstalled git to prevent errors related to cygwin.
         run: Rename-Item "C:\Program Files\Git\usr\bin" notbin
 
+
+      - name: Pre-Build
+        run: 
+          pip install .
```

Comment:
What is this step for?

Reply:
I beleieve it builds gvsbuild? Mostly just following the new instructions updated by wingtk/gvsbuild repo maintainer.

Reply:
Ah so it's because they are using poetry to run the `gvsbuild` command which also avoid the `patches-root-dir` issue. I would suggest doing this instead:

https://github.com/wingtk/gvsbuild/blob/ea97cfebf203b5d4f3b8293fe19adb482c364e75/.github/workflows/ci.yml#L59

Reply:
Im not sure about following that it doesnt follow their own instructions in the readme of how to build also the CI/build for thast commit in their repo failed.